### ISSUE TITLE
Indentation is broken

### ISSUE THREAD

**Issue by RobertAudi** [dag/vim-fish#20]
When I add an `if` statement, and press `<Enter>`, the `end` keyword is added. So far so good. However, if the `if` statement is indented already (or nested), i.e.: The `i` of the `if` and the `e` of `end` is not in column 0, and if I indent the selection with `=` (in normal mode using `==`, or in visual mode), then the indentation completely breaks.

The breakage gets worse every time a block of code is indented, as shown in the screenshots.

![macvim](https://f.cloud.github.com/assets/87562/1688563/8a6f4726-5e0a-11e3-843c-5107e9d56b7b.png)

![macvim](https://f.cloud.github.com/assets/87562/1688561/8a6d601e-5e0a-11e3-9980-eebac6d51468.png)

![macvim](https://f.cloud.github.com/assets/87562/1688562/8a6f1f80-5e0a-11e3-896d-82095b696cca.png)

![macvim](https://f.cloud.github.com/assets/87562/1688560/8a6caa34-5e0a-11e3-9642-0c104da33065.png)

**Comment by alcortesm** [comment]
Vim-fish should not break `=`.

I don't have endwise installed, but my vim is still unable to indent code properly when using vim-fish. 

Endwise is nice but a luxury, indenting code properly is a must.

How can I tell vim-fish to be compatible with `=`?

**Comment by KevinSjoberg** [comment]
Yeah, I'm also seeing broken behaviour when indenting using `=`. Have anyone come to any conclusions regarding this?

**Comment by nhooyr** [comment]
Any updates?

**Comment by c02y** [comment]
@nhooyr None of them works well on indentation.
